<a href="https://colab.research.google.com/github/felixzhao/stat501-project/blob/main/notebooks/2_3_FZ_diff_analysis_citi_per_reg_type.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import os
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:

from google.colab import drive
drive.mount('/content/drive')
!ls "/content/drive/My Drive/Study@NZ/VUW@NZ/STATS 501/Project/csv"

Mounted at /content/drive
citizen_advisor_view.csv     registrations_per_goal.csv
daily_chat_activity.csv      retention_milestone.csv
goals_and_registrations.csv  Sheet1.csv
intervention_history.csv     start_end_registrations.csv
onboarding_medical_data.csv  user_data.csv
onboarding_status.csv	     weight_milestones.csv


# load dfs

In [7]:
se_reg_df_path = '/content/drive/My Drive/Study@NZ/VUW@NZ/STATS 501/Project/pre_process_round_1/start_end_registrations.csv'
reg_per_goal_df_path = '/content/drive/My Drive/Study@NZ/VUW@NZ/STATS 501/Project/csv/registrations_per_goal.csv'
goal_reg_df_path = '/content/drive/My Drive/Study@NZ/VUW@NZ/STATS 501/Project/csv/goals_and_registrations.csv'

## load se reg

In [8]:
se_reg_df = pd.read_csv(se_reg_df_path, low_memory=False)

## load reg per goal

In [10]:
# Assuming table_path contains the path to your CSV file
reg_per_goal_df = pd.read_csv(reg_per_goal_df_path, low_memory=False)

In [11]:
# Convert to DateTime format

datetime_cols = ['lastActivity', 'startGoalDate', 'endGoalDate', 'firstRegistrationDateTime', 'lastRegistrationDateTime']
reg_per_goal_df[datetime_cols] = reg_per_goal_df[datetime_cols].apply(pd.to_datetime, errors='coerce')

# Convert to Category format
category_cols = ['type', 'category', 'goalSubject', 'goalDescription', 'timeAggregation']
reg_per_goal_df[category_cols] = reg_per_goal_df[category_cols].astype('category')

# Convert object columns that should be numerical to float or integer
# If conversion fails, invalid parsing will be set as NaN
numerical_cols = ['goalValue', 'registrationValueAvg', 'registrationValueMin', 'registrationValueMax']
for col in numerical_cols:
    if col in reg_per_goal_df.columns:
        reg_per_goal_df[col] = pd.to_numeric(reg_per_goal_df[col], errors='coerce')
    else:
        print(f"Warning: Column {col} does not exist in DataFrame.")

## load goal and reg

In [12]:
# Assuming table_path contains the path to your CSV file
goal_reg_df = pd.read_csv(goal_reg_df_path, low_memory=False)

# Convert to DateTime format
datetime_cols = ['lastActivity', 'startGoalDate', 'endGoalDate', 'registrationDateTime', 'time']
goal_reg_df[datetime_cols] = goal_reg_df[datetime_cols].apply(pd.to_datetime, errors='coerce')

# Convert to Category format
category_cols = ['type', 'category', 'goalSubject']
goal_reg_df[category_cols] = goal_reg_df[category_cols].astype('category')

# analysis of Citi has goal

In [13]:
tables_dict = {"start_end_registrations": se_reg_df, "registrations_per_goal": reg_per_goal_df, "goals_and_registrations": goal_reg_df}

In [15]:
unique_citi_has_goal = {}
for k, v in tables_dict.items():
    unique_citi = v['citizenId'].unique()
    print(f"there are {len(unique_citi)} unique citizen has goal in {k}")
    unique_citi_has_goal[k] = unique_citi

there are 357 unique citizen has goal in start_end_registrations
there are 362 unique citizen has goal in registrations_per_goal
there are 362 unique citizen has goal in goals_and_registrations


In [25]:
# nested_list = [[1,2,3],[2,3,4],[5,6,7,8]]

nested_list = []
for k in unique_citi_has_goal.keys():
    print(k)
    nested_list.append(set(unique_citi_has_goal[k]))

# unique_elements = []

# for i in range(len(nested_list)):
#     s1 = set(nested_list[i])
#     s2 = set([elem for sublist in nested_list[:i] + nested_list[i+1:] for elem in sublist])
#     unique_elements.append(s1 - s2)

# print(unique_elements)

start_end_registrations
registrations_per_goal
goals_and_registrations


In [30]:
nested_list[1] - nested_list[0]

{0, 831841, 909811, 955121, 1022212}

In [31]:
nested_list[2] - nested_list[0]

{0, 831841, 909811, 955121, 1022212}

In [32]:
nested_list[2] - nested_list[1]

set()

In [33]:
nested_list[1] - nested_list[2]

set()

# Conclusion

There are 4 citizens not in "start end registration", but in "registrations per goal" and "goal and registration" which are "831841, 909811, 955121, 1022212"

the citizens in "registrations per goal" and "goal and registration" are same, count is 362